src/auth/router.js

```diff
@@ -0,0 +1,80 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+const jsonwebtoken = require('jsonwebtoken');
+
+const db = require('../database/connection');
+
+const router = express.Router();
+
+const executeQuery = ({ query, params }) => {
+  return new Promise((resolve, reject) => {
+    db.query(query, [params], (err, rows) => {
+      if (err) {
+        reject(err);
+      }
+
+      if (rows.length === 0) {
+        resolve(null);
+      }
+
+      resolve(rows[0]);
+    });
+  });
+};
+
+const getUserCredentials = ({ id, username }) => {
+  const jwtData = {
+    id,
+    username,
+  };
+
+  const jwtSecret = process.env.JWT_SECRET;
+
+  return jsonwebtoken.sign(jwtData, jwtSecret, {
+    expiresIn: process.env.JWT_EXPIRE_TIME,
+  });
+};
+
+// Read all users with query params
+router.post('/login', async function (req, res) {
+  try {
+    const { username, password } = req.body;
+
+    const user = await executeQuery({
+      query: 'SELECT * FROM user WHERE username = ?',
+      params: username,
+    });
+
+    if (!user) {
+      return res.status(404).json({
+        message: 'User not found',
+      });
+    }
+
+    const isPasswordValid = await bcrypt.compare(password, user.password);
+
```

Comment:
How about implement this by crypto? Bcrypt hide the detail of implemetation, try hashing password and compare by crypto